README.md

```diff
@@ -1,4 +1,166 @@
+
+```python
+ll_default_config = ll.config.get_default_config()
+print(ll_default_config)
+
+ll.config.save_json(ll_default_config, "my_project_ll_config.json")
+```
+More information about the description of the different fields can be found [here](./doc/config.md).
+
+The PyLossless pipeline expects the EEG recordings to be stored as BIDS data. We can demonstrate the usage of the pipeline on a BIDS dataset loaded from OpenNeuro. First, we need to download the dataset
+
+```python
+#Shamelessly copied from https://mne.tools/mne-bids/stable/auto_examples/read_bids_datasets.html
+#pip install openneuro-py
+import openneuro
+from mne.datasets import sample
+
+dataset = 'ds002778'
+subject = 'pd6'
+
+# Download one subject's data from each dataset
+bids_root = sample.data_path() / dataset
+bids_root.mkdir(exist_ok=True)
+
+openneuro.download(dataset=dataset, target_dir=bids_root,
+                   include=[f'sub-{subject}'])
+```
+
+Now that we have a BIDS dataset saved locally, we can use `mne_bids` to load this dataset as as `mne.io.Raw` instance
+
+```python
+import mne_bids
+
+datatype = 'eeg'
+session = 'off'
+task = 'rest'
+suffix = 'eeg'
+bids_path = mne_bids.BIDSPath(subject=subject, session=session, task=task,
+                              suffix=suffix, datatype=datatype, root=bids_root)
+
+raw = mne_bids.read_raw_bids(bids_path)
+```
+
+Great! We have our two ingredients (a dataset and a configuration file), and we can now run the pipeline on that dataset (actually, just one recording in that case)
+
+```python
+pipeline = ll.LosslessPipeline('my_project_ll_config.json')
+pipeline.run(raw)
+
+```
+
+Note that running the pipeline for a full dataset is not much more complicated. We only need a list of `BIDSPath` for all the recordings of that dataset. For example, if `bids_paths` contains such a list, the whole dataset can be processed as follows:
+
+```
+pipeline.run_dataset(bids_paths)
+```
+
+This function essentially loads one raw instance after another from the BIDS recordings specified in `bids_paths` and calls `pipeline.run(raw)` with these raw objects.
+
+
+
+## BIDSification
+
+PyLossless provides some functions to help the user import non-BIDS recordings. Since the code to import datasets recorded in different formats and with different properties can vary much from one project to the next, the user must provide a function that can load and return a `raw` object along with the standard MNE `events` array and `event_id` dictionary. For example, in the case of our dataset
+
+```python
+# Example of importing function
+import tempfile
+def egi_import_fct(path_in, stim_channel):
+
+    # read in a file
+    raw = mne.io.read_raw_egi(path_in, preload=True)
+
+    # events and event IDs for events sidecar
+    events = mne.find_events(raw, stim_channel=['STI 014'])
+    event_id = raw.event_id
+
+    # MNE-BIDS doesn't currently support RawMFF objects.
```

Comment:
We just added this feature into MNE-BIDS : ) but it's probably only available on the bleeding edge github version right now.

Reply:
Well, whenever some features become available in MNE or MNE-BIDS, we can integrate them and drop our own code to avoid duplicating functionalities. Plus, if this code is maintained by other projects, that is that much less to maintain.